vxquery-core/src/main/java/org/apache/vxquery/serializer/XMLSerializer.java

```diff
@@ -441,6 +446,34 @@ private void printSequence(PrintStream ps, SequencePointable seqp, String betwee
         }
     }
 
+    private void printArrayNode(PrintStream ps, TaggedValuePointable tvp) {
+        ArrayPointable ap = pp.takeOne(ArrayPointable.class);
+
+        try {
+            tvp.getValue(ap);
+            if (tvp.getTag() == ValueTag.ARRAY_TAG) {
+                tvp.getValue(ap);
+                int len = ap.getEntryCount();
+                ps.append('[');
+                ps.append(' ');
+                for (int i = 0; i < len; i++) {
+                    ap.getEntry(i, tvp);
+                    print(tvp.getByteArray(), tvp.getStartOffset(), tvp.getLength(), ps);
+                    if (i != len - 1) {
+                        ps.append(',');
+                        ps.append(' ');
+                    }
+                }
+                ps.append(' ');
+                ps.append(']');
+            }
+
+        } finally {
+            pp.giveBack(ap);
+            pp.giveBack(tvp);
```

Comment:
I think that the tvp should be returned by the code that took it from the PointablePool.